plugin/storage/cassandra/spanstore/reader.go

```diff
@@ -142,29 +191,41 @@ func (s *SpanReader) GetTrace(traceID model.TraceID) (*model.Trace, error) {
 	return s.readTrace(dbmodel.TraceIDFromDomain(traceID))
 }
 
-// FindTraces queries for traces.
-func (s *SpanReader) FindTraces(queryParams *spanstore.TraceQueryParameters) ([]*model.Trace, error) {
-	queries, err := BuildQueries(queryParams)
-	if err != nil {
+func validateQuery(p *spanstore.TraceQueryParameters) error {
+	if p == nil {
+		return ErrMalformedRequestObject
+	}
+	if p.ServiceName == "" && len(p.Tags) > 0 {
+		return ErrServiceNameNotSet
+	}
+	if !p.StartTimeMin.IsZero() && !p.StartTimeMax.IsZero() && p.StartTimeMax.Before(p.StartTimeMin) {
+		return ErrStartTimeMinGreaterThanMax
+	}
+	if p.DurationMin != 0 && p.DurationMax != 0 && p.DurationMin > p.DurationMax {
+		return ErrDurationMinGreaterThanMax
+	}
```

Comment:
current implementation has a limitation that duration-based query cannot be combined with tags query, so this should be represented by an explicit error